src/tallies/tally_scoring.cpp

```diff
@@ -2468,4 +2465,63 @@ void score_surface_tally(Particle& p, const vector<int>& tallies)
     match.bins_present_ = false;
 }
 
+void score_pulse_height_tally(Particle& p, const vector<int>& tallies)
+{
+
+  for (auto i_tally : tallies) {
+    auto& tally {*model::tallies[i_tally]};
+
+    // Initialize an iterator over valid filter bin combinations.  If there are
+    // no valid combinations, use a continue statement to ensure we skip the
+    // assume_separate break below.
+    auto filter_iter = FilterBinIter(tally, p);
+
+    auto i_energy_filt = tally.filters()[tally.energy_filter_];
+    auto i_energy_bin = p.filter_matches(i_energy_filt).i_bin_;
+    auto bin_energy = p.filter_matches(i_energy_filt).bins_[i_energy_bin];
+
+    const EnergyFilter& energy_filt {
+      *dynamic_cast<EnergyFilter*>(model::tally_filters[i_energy_filt].get())};
+
+    auto i_cell_filt = tally.filters()[tally.cell_filter_];
+    auto i_cell_bin = p.filter_matches(i_cell_filt).i_bin_;
+    auto bin_cell = p.filter_matches(i_cell_filt).bins_[i_cell_bin];
+
+    const CellFilter& cell_filt {
+      *dynamic_cast<CellFilter*>(model::tally_filters[i_cell_filt].get())};
+
+    const auto& cells = cell_filt.cells();
+    int cell_counter = 0;
+    for (auto cell_id : cells) {
+      p.filter_matches(i_cell_filt).bins_[i_cell_bin] = cell_counter;
+      double score = p.pht_storage()[cell_id];
+
+      if (score < energy_filt.bins().front() ||
+          score > energy_filt.bins().back()) {
+        continue;
```

Comment:
Looks like this will miss `cell_counter++`

Reply:
yes, thats right

Reply:
definitely! Thanks!